lib/puppet/functions/foreman_proxy_content/hostname_from_url.rb

```diff
@@ -0,0 +1,19 @@
+require 'uri'
+
+# @summary Parses a URL and returns the hostname
+#
+# @example Calling the function
+#   $hostname = hostname('https://test.example.com')
+Puppet::Functions.create_function(:'foreman_proxy_content::hostname_from_url') do
+  # @param url URL to parse
+  # @return The hostname found within the URL
+  dispatch :hostname do
+    param 'Stdlib::HTTPUrl', :url
+    return_type 'Stdlib::Host'
+  end
+
+  def hostname(url)
+    uri = URI(url)
+    uri.hostname
```

Comment:
Interesting that https://ruby-doc.org/stdlib-2.7.2/libdoc/uri/rdoc/URI.html makes no mention of `hostname`. There is only host.